image_get/worker/worker/consumer.py

```diff
@@ -0,0 +1,103 @@
+import worker.settings as settings
+import logging as log
+import asyncio
+import aiohttp
+import boto3
+import botocore.client
+import time
+from functools import partial
+from timeit import default_timer as timer
+from worker.util import kafka_connect, parse_message, save_thumbnail_s3, process_image
+
+
+def poll_consumer(consumer, batch_size):
+    """
+    Poll the Kafka consumer for a batch of messages and parse them.
+    :param consumer:
+    :param batch_size: The number of events to return from the queue.
+    :return:
+    """
+    batch = []
+    # Consume messages until either batch_size has been reached or the max
+    # wait time has occurred.
+    max_wait_seconds = 3
+    elapsed_time = 0
+    last_msg_time = timer()
+    msg_count = 0
+    while msg_count < batch_size and elapsed_time < max_wait_seconds:
+        message = consumer.consume(block=False)
+        if message:
+            parsed = parse_message(message)
+            batch.append(parsed)
+            last_msg_time = timer()
+            msg_count += 1
+        elapsed_time = timer() - last_msg_time
+    return batch
+
+
+async def consume(consumer, image_processor, terminate=False):
```

Comment:
I think it would be good to document the structure of the message being consumed somewhere.

Reply:
Thanks, I've described the structure in the README. I expect it to change more as this goes along.